### PR TITLE
Improved documentation for abstractBatchSystem (resolves #744)

### LINKED ISSUE #744: Fix batch System Interface docs
docs/batchSystem "Implementing the batch system interface tutorial"

### PR DESCRIPTION
**Pull request by cket**

### CODE REVIEW

**Comment by cket** [comment]
I added sphinx/rst style parameter, return value, and exception descriptions for the methods and objects in AbstractBatchSystem but no type hints- should those be added?

**Comment by cket** [comment]
Jenkins, test this please.

File: src/toil/batchSystems/abstractBatchSystem.py [hunk 1]
```diff
@@ -39,29 +40,53 @@ def supportsHotDeployment():
...
         """
         return False
 
     def __init__(self, config, maxCores, maxMemory, maxDisk):
-        """This method must be called.
-        The config object is setup by the toilSetup script and
-        has configuration parameters for the jobtree. You can add stuff
-        to that script to get parameters for your batch system.
+        """
+        Initializes initial state of the object
```
Comment (benedictpaten) [inline src/toil/batchSystems/abstractBatchSystem.py]: This comment is a little unnecessary ;)

**Comment by benedictpaten** [comment]
Nice job!!

**Pull request merged**